### ISSUE TITLE
Stabilize render-time sensitive e2e tests

### ISSUE THREAD

**Issue by bkimminich** [juice-shop/juice-shop#183]
Some e2e tests still fails occasionally, supposedly because rendering takes longer than protractor is waiting by default for the subsequent actions. Known affected tests are:
- `challenge "fiveStarFeedback"`
- `challenge "christmasSpecial"`

Most promising solution is replacing the flaky `browser.delay` calls with proper use of `protractor.ExpectedConditions` (available since Protractor 1.7)

Note: This is a follow-up task to #178.
##
